src/ray/common/asio/instrumented_io_context.cc

```diff
@@ -56,7 +56,7 @@ std::string to_human_readable(int64_t duration) {
 }  // namespace
 
 void instrumented_io_context::post(std::function<void()> handler,
-                                   const std::string &name) {
+                                   const std::string name) {
```

Comment:
Does this fix the java segfaults? We should be trying to enable the flag by default right?

Reply:
What I found is sometimes it shows me the garbage names like this when the components are terminated; 
```
	?)????.grpc_server.PinObjectIDs.reply - 1 total (0 active), CPU time: mean = 34.000 ns, total = 34.000 ns
```

But I am not sure if this is related to the Java error.

Reply:
Unless I'm misunderstanding the Windows segfault debugging information, the Java segfaults were happening in a mutex locking frame, so that doesn't seem likely?
```
# Problematic frame:
# C  [libcore_worker_library_java.so+0xa5b141]  absl::lts_2019_08_08::Mutex::Lock()+0x1
```

https://travis-ci.com/github/ray-project/ray/jobs/491603201

Reply:
I assume it is a different issue (but maybe the root cause of both issue is the destruction order).